test/integration/submission/submission.spec.ts

```diff
@@ -755,6 +755,91 @@ describe('Submission Api', () => {
     });
   });
 
+  describe('icgc import', function() {
+    this.beforeEach(async () => {
+      await clearCollections(dburl, ['donors', 'counters']);
+    });
+
+    it('should import legacy samples file', async () => {
+      let file: Buffer;
+      let rows: any[];
+      try {
+        // this file contains 39 rows (samples)
+        // 35 unique donors
+        // 38 unique specimens
+        file = fs.readFileSync(stubFilesDir + `/paca.icgc.samples.tsv`);
+        rows = (await TsvUtils.tsvToJson(stubFilesDir + `/paca.icgc.samples.tsv`)) as any;
+      } catch (err) {
+        return err;
+      }
+
+      const response = await chai
+        .request(app)
+        .post('/submission/icgc-import/preprocess/ABCD-EF')
```

Comment:
didn't notice before but we don't verify the program name during import eh "ABCD-EF" !== "PACA-CA" :stuck_out_tongue: . But I guess that makes sense if a program name is different in argo then in icgc. Maybe a word of caution in swagger when using this endpoint or maybe it should accept the current program name and the program name to use...??

Reply:
This is not critical just something that could happen.

Reply:
so in this pre process endpoint they specify the new program name or fill in the same name.

then in the import they can import whatever they want, we don't need to verify anything because this completely new data. the pre process and import are stateless

Reply:
:+1: True, this endpoint is admins only so I don't think this is troubling.